File: backend/app/types/enums.py
from enum import Enum

class ClubType(Enum):
    COMPLETELY_NEW = 'completely-new'
    ALREADY_EXISTS = 'already-exists'

class DataSource(str, Enum):
    FOOTBALL_ASSOCIATION: str = 'football-association'
    MANUAL: str = 'manual'

class Sport(str, Enum):
    FOOTBALL: str = 'football'

class Result(str, Enum):
    WIN: str = 'W'
    DRAW: str = 'D'
    LOSS: str = 'L'

class HomeAwayNeutral(str, Enum):
    HOME: str = 'H'
    AWAY: str = 'A'
    NEUTRAL: str = 'N'

class Metric(str, Enum):
    APPEARANCES: str = 'Appearances'
    OVERALL_GOALS: str = 'Overall Goals'
    FEATURED_PLAYER: str = 'Featured Player'
    GOALS: str = 'Goals'
    ASSISTS: str = 'Assists'
    PLAYER_OF_MATCH: str = 'Player Of Match'
    BENCH_UNUSED: str = 'Bench Unused'
    POTM: str = 'POTM'
    HATTRICKS: str = 'Hattricks'
    CONSECUTIVE_APPS: str = 'Consecutive Matches Played'
    CONSECUTIVE_WINS: str = 'Consecutive Wins'
    CONSECUTIVE_GOALSCORING_MATCHES: str = 'Consecutive Goalscoring Matches'
    CONSECUTIVE_HATTRICKS: str = 'Consecutive Hattricks'
    POINTS_PER_GAME: str = 'Player Impact - Points Per Game'
    GOALS_SCORED: str = 'Player Impact - Goals Scored'
    GOALS_CONCEDED: str = 'Player Impact - Goals Conceded'
    GOAL_DIFFERENCE: str = 'Player Impact - Goal Difference'
    CLEAN_SHEETS: str = 'Clean Sheets'
    X_SHREK: str = 'xShrek'
    DAYS_BETWEEN_APPS: str = 'Days Between Appearances'
    IMPACT_GOALS: str = 'Impact Goals'
    IMPACT_GOAL_RATIO: str = 'Impact Goal Ratio'

class SplitByType(str, Enum):
    NA: str = 'N/A'
    TOTAL: str = 'Total'
    OPPOSITION: str = 'Opposition'
    PLAYER_COUNT: str = 'Player Count'
    SEASON: str = 'Season'
    YEAR: str = 'Year'
    MONTH: str = 'Month'
    WITH_OR_WITHOUT: str = ''
    KO_TIME: str = 'KO Time'
    MONTH_AND_YEAR: str = 'Month & Year'

class LeaderboardType(str, Enum):    
    APPEARANCES: str = 'Appearances'
    APPEARANCES_BY_SEASON: str = 'Appearances By Season'
    APPEARANCES_BY_YEAR: str = 'Appearances By Year'
    GOALS: str = 'Goals'
    GOALS_PER_GAME: str = 'Goals Per Game'

class MiscStrings(str, Enum):
    WITH: str = 'With'
    WITHOUT: str = 'Without'
    OWN_GOALS: str = 'OWN GOALS'

File: frontend/src/types/enums.ts
export enum CLUB_TYPE {
    COMPLETELY_NEW = 'completely-new',
    ALREADY_EXISTS = 'already-exists'
}

export enum DATA_SOURCE {
    FOOTBALL_ASSOCIATION = 'football-association',
    MANUAL = 'manual'
}

export enum PAGE_TYPE {
    LOGIN = 'Login',
    REGISTER = 'Register'
}

export enum SPLIT_BY_TYPE {
    NA = 'N/A',
    TOTAL = 'Total',
    OPPOSITION = 'Opposition',
    PLAYER_COUNT = 'Player Count',
    SEASON = 'Season',
    YEAR = 'Year',
    MONTH = 'Month',
    KO_TIME = 'KO Time',
}

export enum METRIC {
    APPEARANCES = 'Appearances',
    GOALS = 'Goals',
    HATTRICKS = 'Hattricks',
    POTM = 'POTM',
    CONSECUTIVE_APPS = 'Consecutive Matches Played',
    CONSECUTIVE_WINS = 'Consecutive Wins',
    CONSECUTIVE_GOALSCORING_MATCHES = 'Consecutive Goalscoring Matches',
    CONSECUTIVE_HATTRICKS = 'Consecutive Hattricks',
    POINTS_PER_GAME = 'Player Impact - Points Per Game',
    GOALS_SCORED = 'Player Impact - Goals Scored',
    GOALS_CONCEDED = 'Player Impact - Goals Conceded',
    GOAL_DIFFERENCE = 'Player Impact - Goal Difference',
    CLEAN_SHEETS = 'Clean Sheets',
    X_SHREK = 'xShrek',
    DAYS_BETWEEN_APPS = 'Days Between Appearances',
    IMPACT_GOALS = 'Impact Goals',
}

export enum MANUAL_DATA_ENTRY_ACTION_TYPE {
    ADD_NEW_LEAGUE = 'Add New League & Season',
    ADD_NEW_SEASON = 'Add New Season',
    ADD_NEW_MATCH = 'Add New Match',
    EDIT_MATCH = 'Edit Match'
}

export enum MATCH_LOCATION_TYPE {
    NEW_LOCATION = '-- New Location --'
}

export enum MATCH_COMPETITION_TYPE {
    NEW_COMPETITION = '-- New Competition --'
}

export enum UPDATE_MATCH_SECTIONS {
    MATCH_INFO = 'Match Info',
    PLAYERS = 'Players',
    GOALS_AND_POTM = 'Goals & POTM',
    MATCH_REPORT = 'Match Report'
}

export enum PLAYER_LIST_TYPE {
    ALL_PLAYERS = 'All Club Players',
    ACTIVE_PLAYERS = 'Active Players'
}

export enum MATCH_REPORT_TYPES {
    NO_ALREADY_EXISTS = "No, I'm happy with the already uploaded match report image(s)",
    YES_TEXT = 'Yes, in text form',
    YES_IMAGE = 'Yes, in image form',
    NO = 'No'
}